osu.Game/Screens/Play/Player.cs

```diff
@@ -350,18 +454,20 @@ public override bool OnExiting(IScreen next)
                 return true;
             }
 
-            if ((!AllowPause || HasFailed || !ValidForResume || PausableGameplayContainer?.IsPaused.Value != false || DrawableRuleset?.HasReplayLoaded.Value != false) && (!PausableGameplayContainer?.IsResuming ?? true))
+            if (LoadedBeatmapSuccessfully && canPause)
```

Comment:
```suggestion
            if (canPause)
```

`LoadedBeatmapSuccessfully` is a precondition in `canPause`.